### PR TITLE
Fix test_frame_apply_batch_without_shortcut.

### PR DESCRIPTION
**Pull request by ueshin**
Fix `test_frame_apply_batch_without_shortcut` which could be flaky.

### CODE REVIEW

File: databricks/koalas/tests/test_categorical.py [hunk 1]
```diff
@@ -335,9 +338,10 @@ def to_category(pdf) -> ret:
             return pdf.astype(dtype)
 
         self.assert_eq(
-            kdf.koalas.apply_batch(to_category).sort_index(), to_category(pdf).sort_index()
+            kdf.koalas.apply_batch(to_category).sort_values(["a", "b"]).reset_index(drop=True),
+            to_category(pdf).sort_values(["a", "b"]).reset_index(drop=True),
         )
-
+
```
Comment (xinrong-databricks) [inline databricks/koalas/tests/test_categorical.py:344]: I guess it's fine?

**Review by xinrong-databricks — APPROVED** [review]

File: databricks/koalas/tests/test_categorical.py (hunk 1 above)
Comment (xinrong-databricks) [reply databricks/koalas/tests/test_categorical.py:344]: Probably not..

Comment (ueshin) [reply databricks/koalas/tests/test_categorical.py:344]: oops, removed.

**Comment by xinrong-databricks** [comment]
Format issue, otherwise LGTM!

**Pull request merged**